tests/test_coordinate_source.py

```diff
@@ -0,0 +1,328 @@
+                    box.write(
+                        f"{lower_left_corners[0]}\t{lower_left_corners[1]}\t256\t256\n"
+                    )
+            # populate the second box file with nonsquare coordinates
+            with open(self.box_fp_nonsquare, "w") as box_nonsquare:
+                for center in _centers:
+                    # make a bad box file with non square particles
+                    lower_left_corners = (center[0] - 128, center[1] - 128)
+                    box_nonsquare.write(
+                        f"{lower_left_corners[0]}\t{lower_left_corners[1]}\t256\t100\n"
+                    )
+
+        # Now construct a simulated Relion directory
+        # to test RelionCoordinateSource
+        # The autopick.star file points to sample1.mrc
+        # (created earlier) and relion_coord.star as the coord file
+        os.makedirs(os.path.join(self.data_folder, "AutoPick/job006/"))
+        self.autopick_star_path = os.path.join(
+            self.data_folder, "AutoPick/job006/sample_relion_autopick.star"
+        )
+        shutil.copyfile(
+            os.path.join(self.test_dir_root, "sample_relion_autopick.star"),
+            self.autopick_star_path,
+        )
+        shutil.copyfile(
+            os.path.join(self.test_dir_root, "sample_relion_coord.star"),
+            # renamed because later we glob for sample*star and we don't want this one
+            os.path.join(self.data_folder, "relion_coord.star"),
+        )
+
+        # create default object from a .box file, for comparisons in tests
+        # also provides an example of how one might use this in a script
+        self.all_mrc_paths = sorted(glob(self.data_folder + "/*.mrc"))
+        self.all_box_paths = sorted(glob(self.data_folder + "/sample*.box"))
+        self.files_box = list(zip(self.all_mrc_paths, self.all_box_paths))
+        self.src_from_box = EmanCoordinateSource(self.files_box)
+        # create file lists that will be used several times
+        self.files_coord = list(
+            zip(self.all_mrc_paths, sorted(glob(self.data_folder + "/sample*.coord")))
+        )
+        self.files_star = list(
+            zip(self.all_mrc_paths, sorted(glob(self.data_folder + "/sample*.star")))
+        )
+        self.files_box_nonsquare = list(
+            zip(self.all_mrc_paths, sorted(glob(self.data_folder + "/nonsquare*.box")))
+        )
+
+    def tearDown(self):
+        self.tmpdir.cleanup()
+
+    def testLoadFromCoord(self):
+        # ensure successful loading from particle center files (.coord)
+        CentersCoordinateSource(self.files_coord, particle_size=256)
+
+    def testLoadFromStar(self):
+        # ensure successful loading from particle center files (.star)
+        CentersCoordinateSource(self.files_star, particle_size=256)
+
+    def testLoadFromRelionAutoPick(self):
+        # Here we test loading from a Relion project directory's STAR
+        # index file. We are not using the simulated data source constructed
```

Comment:
`testImages` below suggests differently?

Reply:
Very true, I removed the comment. In the setup it's clear that it's referencing the same data.